src/datepicker/ngb-calendar.ts

```diff
@@ -0,0 +1,61 @@
+import {NgbDate} from './ngb-date';
+import {Injectable} from '@angular/core';
+
+/**
+ * Default implementation of a Gregorian calendar
+ */
+@Injectable()
+export class NgbCalendar {
```

Comment:
We should have an abstract class `NgbCalendar` and the concentrate class `NgbGregorianCalendar` that extends it. We should then only use `NgbCalendar` in DI / other parts of the code.